### PR TITLE
LPS-175992 | master

### PR DESCRIPTION
**Pull request by davisaints**
**JIRA Ticket: [LPS-175992](https://liferay.atlassian.net/browse/LPS-175992)**

### CODE REVIEW

**Comment by davisaints** [comment]
ci:test:relevant

**Comment by davisaints** [comment]
ci:test:objects

**Comment by marcelabc** [comment]
Hey @carolmariaabb 
Could you review this PR, please? :smile:

**Comment by davisaints** [comment]
Comparisons:

- [Relevant x acceptance](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=2473519076&testrayRunIdA=2473544254&statusB=2)

- [Objects suite in this pull request x objects suite](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=2473206253&testrayRunIdA=2473547634&statusB=2)

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by davisaints** [comment]
ci:test:relevant

**Comment by davisaints** [comment]
ci:test:object

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 1]
```diff
@@ -1011,7 +1011,31 @@ public void testAddObjectEntryWithEncryptedObjectField() throws Exception {
...
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"output"
+					).value(
+						ObjectFieldConstants.BUSINESS_TYPE_DECIMAL
+					).build())
+			).build());
+
+		ObjectField objectField2 = _addCustomObjectField(
```
[the remark this answers is not in the document]
Comment (davisaints) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java]: Done!

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -1046,6 +1070,14 @@ public void testAddObjectEntryWithFormulaObjectField() throws Exception {
...
 
+		Assert.assertEquals(
+			objectEntry.getObjectEntryId() + objectEntry.getObjectEntryId(),
+			MapUtil.getDouble(
+				_objectEntryLocalService.getValues(
+					objectEntry.getObjectEntryId()),
+				"idSum"),
+			0);
+
 		Assert.assertEquals(
```
[the remark this answers is not in the document]
Comment (davisaints) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java]: Done!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -1106,6 +1106,10 @@ public Map<String, Serializable> getValues(ObjectEntry objectEntry)
 				).eq(
 					extensionDynamicObjectDefinitionTable.getPrimaryKeyColumn()
 				)
+			).innerJoinON(
+				ObjectEntryTable.INSTANCE,
+				ObjectEntryTable.INSTANCE.objectEntryId.eq(
+					dynamicObjectDefinitionTable.getPrimaryKeyColumn())
```
[the remark this answers is not in the document]
Comment (davisaints) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: You are right, it is a better approach. It works as expected. Thank you, @carolmariaabb!

**Review by carolmariaabb — APPROVED** [review]
🎉

**Comment by davisaints** [comment]
ci:forward:force

**Comment by davisaints** [comment]
ci:forward:force

**Pull request closed without merging**